### ISSUE TITLE
REX-Ray & ScaleIO Gateway v2.0.0.2 (7120)

### ISSUE THREAD

**Issue by cduchesne** [emccode/rexray#554]
The ScaleIO Gateway v2.0.0.2 (7120) introduces a new web server with very limited SSL cipher choices. Golang's crypto/tls package is not compatible and REX-Ray will not be able to connect. In the meantime, please use the ScaleIO Gateway v2.0.0.1 (6035). Please post here or visit us in the community channel on slack if you need any assistance in getting the earlier version installed.

slack: codecommunity.slack.com #project-rexray

**Comment by cduchesne** [comment]
@akutz has contributed pull request #555 that enhances the crypto/tls package to allow communication with version 2.0.0.2 of the ScaleIO Gateway. When the gateway software is fixed to support additional SSL ciphers, we will revert back to the standard crypto/tls package.

Do note that the custom crypto/tls package will only being used by the ScaleIO driver.

**Comment by cduchesne** [comment]
Upon further investigation, we have found that with ScaleIO Gateway v2.0.0.2, the supported TLS 1.2 ciphers depends on the installed version of java. This is different from v2.0.0.1 in which the TLS implementation was significantly different. 

Oracle's default java installs includes the necessary ciphers for REX-Ray to function properly, however OpenJDK does not. Being that the OpenJDK is commonly deployed, we have made the decision to support PR #555.
